### PR TITLE
Add transaction:sign command - Closes #321

### LINKED ISSUE #321: Add transaction:sign command
### Description

Add transaction:sign command which expects base64 string as argument and sign transaction from single/multi signatures account.
```
transaction:sign GxT7g9YMBOO8wkr0bygKoQz70vRsKmK7PPrKYi4+80q8jr+j95NVQQDvlEIX1L6d6ThBR7D8eZdYEbLFF+aXDw==
```
It should support flags
```
--includeSender (boolean)
--mandatoryKeys (allowMultiple: true) base64 string
--optionalKeys (allowMultiple: true) base64 string
--json
--pretty
```
If `json` is not specified it should return in base64 string format.
If includeSender is true, mandatory or optional keys must exist

### Motivation

A command to sign unsigned/signed transaction 

### Acceptance Criteria

- Test case where mandatory keys and optional keys are not specified
- Test case where mandatory keys are specified
- Test case where optional keys are specified
- Test case where mandatory keys and optional keys are specified
- Test case where mandatory keys, optional keys and include sender are specified

### PR DESCRIPTION
**Pull request by ManuGowda**
### What was the problem?

This PR resolves #321 

### How was it solved?

- Implement sign command to sign transaction from single/multi signatures account.

### How was it tested?

- Added unit tests
- Manual testing
```
./bin/run transaction:sign hz2oWizucNpjHZCw8X+tqMOsm4OyYT9Mpf3dN00QNLM= CAIQABgCIIDC1y8qIPG59O5xtdWFfTs0bUQcqWfyeHDr7ohWnbNk/RPiitujMiQIZBIUqwBBp9P3ssKQtbg01Gvce364WBUaCnNlbmQgdG9rZW4= --mandatory-keys="8bn07nG11YV9OzRtRBypZ/J4cOvuiFads2T9E+KK26M=" --mandatory-keys="SmdkakRjE9uWTDk3A1mEXFL86SJaOSl3DvQUSMJY/Tk=" --optional-keys="+kBraVLTd/AniSDj642pGeTPXGiwLuul2LMzT9wDabY=" --optional-keys="V99cOBGWGTn43PqFjG6u/r+qTelC9+cDv4gSfg7pzKQ=" --passphrase='desk deposit crumble farm tip cluster goose exotic dignity flee bring traffic' --json
```

### CODE REVIEW

**Review by shuse2 — APPROVED** [review]

File: src/commands/transaction/sign.ts [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+ */
+
+import { flags as flagParser } from '@oclif/command';
+import { transactions, codec, TransactionJSON } from 'lisk-sdk';
+import BaseIPCCommand from '../../base_ipc';
+import { flags as commonFlags } from '../../utils/flags';
+import { getPassphraseFromPrompt } from '../../utils/reader';
+
+export default class SignCommand extends BaseIPCCommand {
+	static description = 'Sign a encoded transaction.';
```
Comment (ishantiw) [inline src/commands/transaction/sign.ts:23]:
```suggestion
	static description = 'Sign an encoded transaction.';
```

File: src/commands/transaction/sign.ts [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+
+	static args = [
+		{
+			name: 'networkIdentifier',
+			required: true,
+			description:
+				'Network identifier defined for the network or main | test for the Lisk Network.',
```
Comment (ishantiw) [inline src/commands/transaction/sign.ts:30]:
```suggestion
				'Network identifier defined for a network or main | test for the Lisk Network.',
```

Comment (ishantiw) [reply src/commands/transaction/sign.ts:30]: also, what do you mean by `main | test for the Lisk Network`? It sounds like that either give `networkIdentifier` for a network or give `main | test`

**Review by ishantiw — COMMENTED** [review]
Just a couple of comments

File: src/commands/transaction/sign.ts (hunk 2 above)
Comment (ManuGowda) [reply src/commands/transaction/sign.ts:30]: mmm yea,  this can be confusing, but it exists since the introduction, any suggestion for change?

**Review by ishantiw — APPROVED** [review]

**Pull request merged**